src/common/components/resetPasswordModal/resetPasswordModal.component.js

```diff
@@ -38,9 +41,11 @@ class ResetPasswordModalController {
   }
 
   $onDestroy() {
-    angular.forEach( ['e', 'k'], ( key ) => {
-      this.$location.search( key, null );
-    } );
+    this.removeQueryParams();
+    if(!this.exitWithoutRefresh){
+      // use $timeout here as workaround to Firefox bug
+      this.$timeout(() => this.$window.location.reload());
```

Comment:
This now refreshes the page when the reset modal is closed except when `Back to Sign In` is clicked. Back to sign won't refresh the page if the user signs in there. Maybe that should just be a cancel link that closes the modal and refreshes the page too...